src/lua_interface.c

```diff
@@ -0,0 +1,504 @@
+/***********************************************************************
+ *
+ * Copyright (c) 2021
+ * IoTech Ltd
+ *
+ **********************************************************************/
+#include <stdbool.h>
+#include <assert.h>
+
+#include "lua_interface.h"
+#include "defines.h"
+#include "device.h"
+
+
+static lua_State *lua_state;
+
+static void lua_fail (lua_State *L);
```

Comment:
Could be worth changing 'L' to be a more meaningful name and in line with coding standard ie lowercase